### PR TITLE
refine log and process for distribution handler

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #23085 
/kind enhancement

### CODE REVIEW

File: internal/querycoordv2/balance/utils.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func CreateSegmentTasksFromPlans(ctx context.Context, checkerID int64, timeout t
 
 		log.Info("create segment task",
 			zap.Int64("collection", p.Segment.GetCollectionID()),
+			zap.Int64("segmentID", p.Segment.GetID()),
```
Comment (yah01) [inline internal/querycoordv2/balance/utils.go:70]: Please also add this log field at line 59

**Review by yah01 — COMMENTED** [review]
/lgtm

**Review by yah01 — COMMENTED** [review]
/lgtm

File: internal/querycoordv2/dist/dist_handler.go [hunk 2]
```diff
@@ -225,14 +232,12 @@ func (dh *distHandler) getDistribution(ctx context.Context) error {
 	})
 
 	if err != nil {
-		return err
+		return nil, err
```
Comment (sunby) [inline internal/querycoordv2/dist/dist_handler.go:235]: why this method returns the response back?

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**